Source/misdat.h

```diff
@@ -5,19 +5,104 @@
  */
 #pragma once
 
-#include "missiles.h"
-
 namespace devilution {
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+typedef enum missile_resistance : uint8_t {
+	MISR_NONE,
+	MISR_FIRE,
+	MISR_LIGHTNING,
+	MISR_MAGIC,
+	MISR_ACID,
+} missile_resistance;
+
+typedef enum missile_graphic_id : uint8_t {

```

Comment:
If we're using C++ there is no need for the `typedef` stuff on structs or enums by the way.
It's just `enum A {}; struct B {};`